src/particles/ImpactXParticleContainer.cpp

```diff
@@ -111,19 +110,22 @@ namespace impactx
          * is then copied to the permanent tile which is stored on the particle
          * (particle_tile).
          */
-        using PinnedTile = amrex::ParticleTile<NStructReal, NStructInt, NArrayReal, NArrayInt,
+        using PinnedTile = amrex::ParticleTile<amrex::SoAParticle<RealSoA::nattribs, IntSoA::nattribs>, RealSoA::nattribs, IntSoA::nattribs,
                 amrex::PinnedArenaAllocator>;
+        using SoAParticle = amrex::SoAParticle<RealSoA::nattribs, IntSoA::nattribs>;
+
         PinnedTile pinned_tile;
+        auto ptd = pinned_tile.getParticleTileData();
```

Comment:
Does this have to be a reference?
```suggestion
        auto & ptd = pinned_tile.getParticleTileData();
```